### PR TITLE
[WIP] DGS-5661: moving schema-registry cluster commands to srcm/v2

### PR DESCRIPTION
**Pull request by Anshul-Goyal**
Release Notes
-------------

Breaking Changes
- PLACEHOLDER

New Features
- PLACEHOLDER

Bug Fixes
- PLACEHOLDER

Checklist
---------
1. [CRUCIAL] Is the change for features that are already live in prod?
   * yes: ok
   * no: DO NOT MERGE until the required features are live in prod  

What
----

References
----------

Test & Review
-------------

Open Questions / Follow-ups
---------------------------

### CODE REVIEW

**Comment by brianstrauch** [comment]
Looks like there are lots of similarities between this and https://github.com/confluentinc/cli/pull/1626. Should we wait until one is merged (or branch one off of the other) to avoid code duplication?

File: internal/cmd/schema-registry/command_cluster_delete.go
Comment (brianstrauch) [inline internal/cmd/schema-registry/command_cluster_delete.go:48]: The second argument can contain useful error details. Take a look at some other SDKs for how they do this.

Comment (brianstrauch) [reply internal/cmd/schema-registry/command_cluster_delete.go:48]: https://github.com/confluentinc/cli/blob/main/internal/pkg/ccloudv2/cdx.go is a good example.

Comment (brianstrauch) [reply internal/cmd/schema-registry/command_cluster_delete.go:48]: Based on the function name, it looks like this is returning one cluster, yes?
```suggestion
	cluster, _, err := c.V2Client.GetSchemaRegistryClusterByEnvironment(environmentId)
```

File: internal/cmd/schema-registry/command_cluster_delete.go
Comment (brianstrauch) [inline internal/cmd/schema-registry/command_cluster_delete.go:52]: Is there a better way to check this? Is there an `enabled` field, or should the backend return an error if it's disabled?

File: internal/cmd/schema-registry/command_cluster_describe.go
Comment (brianstrauch) [inline internal/cmd/schema-registry/command_cluster_describe.go:38]: I really wish the SDK was able to define this...

File: internal/cmd/schema-registry/command_cluster_describe.go
Comment (brianstrauch) [inline internal/cmd/schema-registry/command_cluster_describe.go:194]: Nit: Pass arguments in the same order they're used.
```suggestion
func getMaxSchemaLimitPriceKey(serviceProvider, serviceProviderRegion, sgPackage string) string {
```

File: internal/cmd/schema-registry/command_cluster_upgrade.go
Comment (brianstrauch) [inline internal/cmd/schema-registry/command_cluster_upgrade.go:59]: We try to maintain a 1-1 mapping of flag names to variable names. In the long run, it makes the code a bit more maintainable and easier to read.
```suggestion
	package, err := cmd.Flags().GetString("package")
```

File: internal/pkg/ccloudv2/schema-registry.go
Comment (brianstrauch) [inline internal/pkg/ccloudv2/schema-registry.go:42]: Did calling `Execute()` work for you? In the past, other SDKs have failed on this function.

File: test/test-server/srcm_handlers.go
Comment (brianstrauch) [inline test/test-server/srcm_handlers.go:18]: In the real world, do region IDs look like "us-west-2"?

File: test/test-server/srcm_handlers.go
Comment (brianstrauch) [inline test/test-server/srcm_handlers.go:24]: Using a `switch` is preferred. It reduces boilerplate and makes it easier to add new branches.

File: test/test-server/srcm_handlers.go
Comment (brianstrauch) [inline test/test-server/srcm_handlers.go:77]:
```suggestion
	return srcm.SrcmV2ClusterList{
		Data: []srcm.SrcmV2Cluster{getSchemaRegistryCluster(id, packageType, httpEndpoint, SRApiEnvId, regionId, status)},
	}
```

File: test/test-server/srcm_handlers.go
Comment (brianstrauch) [inline test/test-server/srcm_handlers.go:67]: Structs with a single field can be put on one line
```suggestion
		Spec: &srcm.SrcmV2ClusterSpec{
			DisplayName:  &id,
			Package:      &packageType,
			HttpEndpoint: &endpoint,
			Environment: &srcm.GlobalObjectReference{Id: envId},
			Region: &srcm.GlobalObjectReference{Id: regionId},
		},
		Status: &srcm.SrcmV2ClusterStatus{Phase: status},
```

**Review by brianstrauch — CHANGES_REQUESTED** [review]

File: internal/pkg/ccloudv2/schema-registry.go
Comment (sgagniere) [inline internal/pkg/ccloudv2/schema-registry.go:25]: This should set the page size to `ccloudV2ListPageSize`, and probably handle next page tokens, to avoid issues with users going over the default page size limit and getting truncated results.

**Pull request closed without merging**
